### ISSUE TITLE
Add support for alerting on azure datasource

### ISSUE THREAD

**Issue by dkmla** [grafana/azure-monitor-datasource#11]
I need the possibility of enabling alerts from the azure metrics.

Currently I get the following error:
"The datasource does not support alerting queries"

**Comment by daniellee** [comment]
At the moment, a data source has to be part of core Grafana to be able to support alerting. If this gets merged into core, depends on the uptake. So alerting is not even possible for now.

**Comment by regevbr** [comment]
In order to make this data source popular, alerts will play a big part of it... Currently I monitor my message queues with it but with no alerts it has little use to me...

**Comment by sguleff** [comment]
Agree with this.  I work for a large eCommerce company on Azure.  We'd be able to leverage this if only there were alerting available.

**Comment by daniellee** [comment]
The Grafana team is working on making data source plugins for the backend which means that we will be able to add a backend for this plugin and automatically get alerting. But it is still a month or two away in the future.

**Comment by goatmale** [comment]
Any update on this? It looks like I can send test alerts, but I get a big ole error that it's not supported.
Thanks.

**Comment by Sojjocola** [comment]
Any news? 

Thanks
François

**Comment by faesel** [comment]
Has there been any progress on this?

After being frustrated but the lack of flexibility for alerting on the Azure portal, I discovered this plugin and thought it would be the solution to having an alerting system for custom telemetry events.

**Comment by daniellee** [comment]
The azure monitor plugin will be added to core Grafana in the coming month or two (or three). See the Grafana roadmap [here](https://github.com/grafana/grafana/blob/master/ROADMAP.md#mid-term-2-4-months). Once we have reimplemented some of the query logic in Go (Grafana's backend language) then it will automatically have support for alerting.

**Comment by francoisbeaussier** [comment]
Hi @daniellee, any update on this? The roadmap link you supplied does not mention the azure monitor plugin, so maybe the grafana team made the backend changes required for alerting to be implemented?

**Comment by alhardy** [comment]
Hope so, really keen on this too ;)

**Comment by dbronk** [comment]
@daniellee any update on alerting from Azure?  Without alerting, Grafana is little value add for me.

**Comment by daniellee** [comment]
Currently working on adding support for Log Analytics. This added more dependencies which complicates adding it to core Grafana. Still a candidate for being released in Grafana v5.3 (so nothing has changed in the planning so far).

**Comment by dbronk** [comment]
Thanks @daniellee - Any ETA?

**Comment by amitkumarcn** [comment]
@daniellee Any Progress with alerting in azure monitor plugin?

**Comment by bhozar** [comment]
Grafana 5.3 doesn't have many items to complete for the milestone, and I see no mention of Azure Monitoring alerting. Is it still on track for the 5.3 release?

**Comment by cypherhyper** [comment]
What's the latest on this?

**Comment by spurthimallesh** [comment]
Hello, 

Currently I am using Grafana v5.2.4 and I can successfully install the azure-monitor datasource and create a graph for application insights. But I am facing a problem with alert config for the same .


![image](https://user-images.githubusercontent.com/39271577/46003149-b85d6500-c0a7-11e8-8a99-6e9f18d3cee3.png)


Could you please help me with this issue. 

Thanks

**Comment by daniellee** [comment]
The plugin has to be moved into Grafana core to get alerting support. We have focused on adding support for Azure Log Analytics the last few months so moving this plugin into core was postponed. The work to do the move is planned for the next few months and should be included in Grafana 6.0 (with a release in February 2019). But this is only preliminary so nothing guaranteed.

**Comment by arthurvanzyl** [comment]
Hi Everyone, 

Same issue experienced on Azure Application Insights and Grafa.  Is there any work around for this issue ? 
A

**Comment by CarlosTenorio** [comment]
Something new about it? :octocat:

**Comment by javiermarasco** [comment]
Adding into the people waiting for this to come to release. Alerting is really a needed feature.

Thanks!

**Comment by jungopro** [comment]
Same here, I’m using azure alerts now and I really don’t like it (putting it delicately :smile)

**Comment by TecumsehSamp** [comment]
Guys could you please just subscribe for notifications to this thread and not keep posting the same thing? I'm sure I'm not the only one who's subscribed and is getting emails constantly about people asking the same thing. 

We know.. we all want this and their working on it ;)

**Comment by chris-mason-zuto** [comment]
When is alerting likely to be added?

**Comment by Tulkis** [comment]
Hi all,
Is possible get the azure monitor alerts into dashboard? (I'm referring the alerts created in Azure Monitor -Alerts)
Thank for advance.
Eduard.

**Comment by daniellee** [comment]
@Tulkis being worked on right now and will hopefully be included in Grafana 6.0 in 2 weeks time. Have just moved Azure Monitor into Grafana for the first beta release (so will be closing this GitHub repo in the near future and all open issues will be ported over to the main Grafana repo).

**Comment by Tulkis** [comment]
@daniellee excuse me, I'm talking about the whole alerts we have in Azure Portal (service alerts , log search alerts, metric alerts, etc). 
I understand this thread is for the data source metric alerts?

**Comment by daniellee** [comment]
@Tulkis oh sorry. No plans currently for integrating that into Grafana. 

At the moment, I don't have enough insight into how popular the Azure alerts are. Also, until we have completed the project to have a pluggable alert engine for Grafana, it would be a technically difficult feature to add.

**Comment by javiermarasco** [comment]
As of the launch of Grafana 6.0 (as beta) the datasource for Azure is integrated and no longer an external plugin for this reason alerting should be working now.

I am testing grafana 6.0Beta now with alerts on Azure but still getting the "Alerts not supported for this datasource" error, I will continue investigating on what could be wrong for getting this error.

If someone else has the same situation, please reach back to check possibilities.

Thanks!

**Comment by marefr** [comment]
@javiermarasco alerting didn't make it for 6.0-beta1, but will hopefully be included 6.0-beta2 and/or 6.0 stable.

**Comment by kwaazaar** [comment]
So it didn't seem to have made it into 6.0 and [no commits for the last month](https://github.com/grafana/grafana/tree/master/pkg/tsdb/azuremonitor).
Since the roadmap doesn't show much detail, can anyone indicate when alerting can be expected?

**Comment by JamesHealey94** [comment]
> So it didn't seem to have made it into 6.0 (which is not according to [the docs](http://docs.grafana.org/alerting/rules/)) and [no commits for the last month](https://github.com/grafana/grafana/tree/master/pkg/tsdb/azuremonitor).
> Since the roadmap doesn't show much detail, can anyone indicate when alerting can be expected?

@kwaazaar Azure Monitor alerting is in 6.0.0 (I've just created a test alert and it fired as expected).

**Comment by dougnd** [comment]
I believe some other parts of the Azure Monitor datasource are not yet available for alerting (Log Analytics, Application Insights). I saw a bunch of issues get bumped to a future 6.1 release.

https://github.com/grafana/grafana/issues/14943
https://github.com/grafana/grafana/issues/15153

Consider following those issues to get more feedback on when they will be released.

**Comment by kwaazaar** [comment]
I guess you guys are right as I no do see alerts coming in. However, my alerts (currently sending them to an MS Teams-channel) do contain this errormessage: "Error message | tsdb.HandleRequest() error Alerting not supported for Azure Log Analytics", so quite confusing still.

My dashboard query is running on Application Insights, one of the ' in progress' sources mentioned above.

**Comment by javiermarasco** [comment]
Hi Robert, app insights and log search don’t support alerting yet and I
can’t find a roadmap on when they will be supported, for those now you only
have azure monitoring :(

**Comment by stebet** [comment]
Any ETA on App Insights and Azure Log analytics alerting? Is it being worked on?

**Comment by bhozar** [comment]
6.2-beta1 has just been released, but it looks like it has been bumped from 6.2 to the next milestone, 6.3 :(

**Comment by daniellee** [comment]
This repo is going to be archived. The Azure Monitor plugin was moved into the Grafana repository for version 6.0 of Grafana. If you are using Grafana 6.x then you will automatically start using the built-in version of the Azure Monitor datasource.

Announcement: https://grafana.com/blog/2019/02/25/grafana-v6.0-released/#azure-monitor-datasource
Documentation for Azure Monitor in Grafana: https://grafana.com/docs/features/datasources/azuremonitor/

This issue is partially implemented. There are already issues in the Grafana repo for supporting alerting for Azure Logs and for App Insights:

- https://github.com/grafana/grafana/issues/14943
- https://github.com/grafana/grafana/issues/15153

Current status is that I have started on alerting support for App Insights and it will be included in 6.3. It is also on our backlog is to start working on alerting for Azure Logs (as well as the Azure Data Explorer datasource plugin which also uses the Kusto query language). Not sure exactly when that will be done however, 6.3 maybe but more likely to be 6.4 or 6.5.
